src/coreComponents/finiteVolume/mimeticInnerProducts/BdVLMInnerProduct.hpp

```diff
@@ -0,0 +1,217 @@
+
+    if( LvArray::tensorOps::AiBi< 3 >( cellToFaceVec, faceNormal ) < 0.0 )
+    {
+      LvArray::tensorOps::scale< 3 >( faceNormal, -1 );
+    }
+
+    // the two-point transmissibility is computed to computed here because it is needed
+    // in the implementation of the transmissibility multiplier (see below)
+    // TODO: see what it would take to bring the (harmonically averaged) two-point trans here
+    real64 diagEntry = 0.0;
+    MimeticInnerProductHelpers::ComputeInvTPFATransWithMultiplier< NF >( elemPerm,
+                                                                         faceNormal,
+                                                                         faceAreaMat[ifaceLoc][ifaceLoc],
+                                                                         transMultiplier[elemToFaces[ifaceLoc]],
+                                                                         weightToleranceInv,
+                                                                         cellToFaceVec,
+                                                                         diagEntry );
+    tpTransInv[ifaceLoc] = diagEntry;
+
+    normalsMat[ ifaceLoc ][ 0 ] = faceNormal[ 0 ];
+    normalsMat[ ifaceLoc ][ 1 ] = faceNormal[ 1 ];
+    normalsMat[ ifaceLoc ][ 2 ] = faceNormal[ 2 ];
+
+  }
+
+  // 2) compute N of Beirao da Veiga, Lipnikov, Manzini
+  LvArray::tensorOps::Rij_eq_AikBkj< NF, 3, 3 >( work_numFacesByDim,
+                                                 normalsMat,
+                                                 permMat );
+
+  // 3) compute ( N^T R )^-1 of Beirao da Veiga, Lipnikov, Manzini
+  LvArray::tensorOps::Rij_eq_AkiBkj< 3, 3, NF >( work_dimByDim,
+                                                 work_numFacesByDim,
+                                                 cellToFaceMat );
+  LvArray::tensorOps::invert< 3 >( work_dimByDim );
+
+  // 4) compute N ( N^T R )^-1 N^T
+  LvArray::tensorOps::Rij_eq_AikBjk< 3, NF, 3 >( work_dimByNumFaces,
+                                                 work_dimByDim,
+                                                 work_numFacesByDim );
+  LvArray::tensorOps::Rij_eq_AikBkj< NF, NF, 3 >( transMatrix,
+                                                  work_numFacesByDim,
+                                                  work_dimByNumFaces );
+
+  // 5) compute the stabilization coefficient \tilde{ \lambda }
+  real64 const stabCoef = 2.0 / NF * LvArray::tensorOps::trace< NF >( transMatrix );
+
+  // at this point we have N ( N^T R )^-1 N^T and \tilde{ \lambda }
+  // we have to compute I - R ( R^T R )^-1 R^T and sum
+
+  // 6) compute ( R^T R )^-1
+  real64 temp[ NF ][ 3 ] = {{ 0 }};
+  LvArray::tensorOps::copy< NF, 3 >( temp, cellToFaceMat ); // <<-- TODO: ask about this, we could have Rij_eq_AkiAkj instead
+  LvArray::tensorOps::Rij_eq_AkiBkj< 3, 3, NF >( work_dimByDim,
+                                                 temp,
+                                                 cellToFaceMat );
```

Comment:
@corbett5 In this function, I need to compute `R^T R`, where `R` is a rectangular matrix of size `NF x 3`. Currently I create a temporary matrix to be able to use `LvArray::tensorOps::Rij_eq_AkiBkj`. Is there already something in LvArray to do the same thing without the temporary matrix? If not, would it make sense to add something like `Rij_eq_AkiAkj`?

Reply:
I don't think there is currently anything to do this since the `restrict` keyword prevents the usage of `AkiBkj`. But yes given this restriction it would absolutely make sense to add. @rrsettgast and @CusiniM have a PR to do the same thing for other functions.

Reply:
@francoishamon We can quickly add it to that PR (216) if you want.

Reply:
Yes that would be great (if you don't mind). Do you want to do it? Or should I do it and commit to your PR? Thanks

Reply:
I have done it. I just need to add a unit test for it.